### ISSUE TITLE
Weight search results by effective_amount

### ISSUE THREAD

**Issue by kauffj** [lbryio/lighthouse#32]
Prompted by the fact that if I search "disaster", "It's A Disaster" is currently the third result.

**Comment by tiger5226** [comment]
This is a nice improvement. We currently don't have access to `effective_amount` in the elastic document. However, as part of ChainQuery we can ensure this information is sent from there so it can be included in the elastic document to be used as a weight on the search score.

**Comment by tiger5226** [comment]
Once #71 is closed out we can modify the elastic search query to give weight to the effective amount since it will now be coming from [chainquery](https://github.com/lbryio/chainquery).

**Comment by tiger5226** [comment]
solved with https://github.com/lbryio/lighthouse/pull/81#event-1692597622

**Comment by filipnyquist** [comment]
Closed, see comment above

**Comment by tiger5226** [comment]
reopening because while this is fixed in the leading branch it has not been deployed yet.

**Comment by tiger5226** [comment]
A quick update on this one. It was merged in but had to be rolled back. The function which used the effective amount as a factor of the score threw exceptions for certain documents which did not contain a number in the effective amount field. This should not have been the case. I could not reproduce this locally either. So maybe we need to plan a rebuild of the elastic search db in production.  see docs here https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-function-score-query.html

**Comment by alyssaoc** [comment]
@tiger5226 Is this still an outstanding issue? Will Sean's recent changes to search impact this?

**Comment by tiger5226** [comment]
yeah, it is but we need to come up with an alternative solution or fix the production elastic search instance. 

@kauffj how important is this? Do we want to rebuild the production db? It takes about an hour to do it.  So search would be down for an hour. 

@nikooo777 Could we do this on another machine and replace it as lighthouse.lbry.io once complete so search does not go down?

**Comment by nikooo777** [comment]
> @nikooo777 Could we do this on another machine and replace it as lighthouse.lbry.io once complete so search does not go down?

absolutely

**Comment by filipnyquist** [comment]
Get a server and I can set it up!

Den 21 aug. 2018 11:36 skrev "Niko" <notifications@github.com>:

@nikooo777 <https://github.com/nikooo777> Could we do this on another
machine and replace it as lighthouse.lbry.io once complete so search does
not go down?

absolutely

—
You are receiving this because you modified the open/close state.
Reply to this email directly, view it on GitHub
<https://github.com/lbryio/lighthouse/issues/32#issuecomment-414614244>,
or mute
the thread

**Comment by kauffj** [comment]
It looks like you guys already figured out the approach on how to do this that I would have recommended.

It's unclear how much value this will have, but I think it's a smart change. It did make me realize we should figure out ways to ascertain how search is performing and I will file a ticket related to that.
